common/components/GasSlider/components/SimpleGas.tsx

```diff
@@ -3,22 +3,41 @@ import Slider from 'rc-slider';
 import translate from 'translations';
 import { gasPriceDefaults } from 'config/data';
 import FeeSummary from './FeeSummary';
+import { Spinner } from 'components/ui';
+import { CSSTransition } from 'react-transition-group';
+import { State as NetworkState } from 'reducers/transaction/network';
 import './SimpleGas.scss';
 
 interface Props {
   gasPrice: string;
+  gasEstimationStatus: NetworkState['gasEstimationStatus'];
   changeGasPrice(gwei: string): void;
 }
 
 export default class SimpleGas extends React.Component<Props> {
   public render() {
-    const { gasPrice } = this.props;
-
+    const { gasPrice, gasEstimationStatus } = this.props;
+    const estimatingGas = gasEstimationStatus === 'PENDING' ? true : false;
+    const estimatingGasTimeout = gasEstimationStatus === 'TIMEOUT' ? true : false;
     return (
       <div className="SimpleGas row form-group">
-        <div className="col-md-12">
+        <div className="col-md-12 SimpleGas-title">
           <label className="SimpleGas-label">{translate('Transaction Fee')}</label>
+          <div className="SimpleGas-flex-spacer" />
+          <CSSTransition in={estimatingGas} timeout={300} classNames="fade">
+            <div className={`SimpleGas-estimating small ${estimatingGas ? 'active' : ''}`}>
+              Setting gas limit
```

Comment:
Nitpick: I think "Setting" doesn't make a lot of sense here, since that shouldn't take time. Maybe "Calculating" or "Determining" or "Fetching" or something like that.